### PR TITLE
Homogenize sound settings usage and allow zeroed sound volume

### PR DESCRIPTION
**Pull request by StraToN**

### CODE REVIEW

File: addons/escoria-core/game/scenes/sound/esc_music_player.gd [hunk 1]
```diff
@@ -44,10 +44,7 @@ func set_state(p_state: String, p_force: bool = false) -> void:
 			resource.loop_end = resource.mix_rate * resource.get_length()
 		elif "loop" in resource:
 			resource.loop = true
-		if escoria.project_settings_manager.has_setting(escoria.project_settings_manager.MUSIC_VOLUME):
-			stream.volume_db = escoria.project_settings_manager.get_setting(
-				escoria.project_settings_manager.MUSIC_VOLUME
-			)
+		stream.volume_db = escoria.settings.music_volume
```
Comment (dploeger) [inline addons/escoria-core/game/scenes/sound/esc_music_player.gd:47]: I'm confused about this. Isn't using `escoria.project_settings_manager` the correct way or did I miss something? (probably?)

**Review by dploeger — CHANGES_REQUESTED** [review]

File: addons/escoria-core/game/scenes/sound/esc_music_player.gd (hunk 1 above)
Comment (BHSDuncan) [reply addons/escoria-core/game/scenes/sound/esc_music_player.gd:47]: I was actually going to ask the same thing, so maybe we both missed something.

**Comment by balloonpopper** [comment]
I've tested and changing the volume works. I'm still confused that there's 2 different volume setting methods though. e.g. for the sound voume :

apply_settings (escoria.gd) 
```
		AudioServer.set_bus_volume_db(
			AudioServer.get_bus_index(BUS_MUSIC),
			linear2db(settings.music_volume)
		)
```
and
esc_sound_player.gd
```
stream.volume_db = escoria.settings.music_volume
```

**Review by balloonpopper — CHANGES_REQUESTED** [review]
I've tested and changing the volume works. I'm still confused that there's 2 different volume setting methods though. e.g. for the sound voume :

apply_settings (escoria.gd)
```
		AudioServer.set_bus_volume_db(
			AudioServer.get_bus_index(BUS_MUSIC),
			linear2db(settings.music_volume)
		)
```
and
esc_sound_player.gd
```
stream.volume_db = escoria.settings.music_volume
```

Is it possible to have a single function for this to make sure it's done consistently?

**Comment by StraToN** [comment]
It feels that settings management is a bit incoherent here. I'll review the whole thing and update.

File: addons/escoria-core/game/scenes/sound/esc_music_player.gd (hunk 1 above)
Comment (StraToN) [reply addons/escoria-core/game/scenes/sound/esc_music_player.gd]: Fixed

**Comment by balloonpopper** [comment]
Noting this here so I remember what to retest :
 If you "say" something, and change volume before the audio file finishes, the audio file is stuck on repeat.

**Comment by StraToN** [comment]
> Noting this here so I remember what to retest : If you "say" something, and change volume before the audio file finishes, the audio file is stuck on repeat.

I confirm the issue.
This can only be resolved by re-importing all audio files in res://speech/ with `loop` flag **disabled**.

I documented this new requirement in https://github.com/godot-escoria/escoria-docs/pull/53

**Review by BHSDuncan — APPROVED** [review]

**Comment by StraToN** [comment]
Superseded by #616

**Pull request closed without merging**